qiskit_experiments/characterization/t1_experiment.py

```diff
@@ -0,0 +1,207 @@
+                ),
+            }
+        )
+
+        if unit == "dt":
+            analysis_result["fit"]["dt"] = conversion_factor
+            analysis_result["fit"]["circuit_unit"] = unit
+
+        return analysis_result, None
+
+    @staticmethod
+    def _fit_quality(fit_out, fit_err, reduced_chisq):
+        # pylint: disable = too-many-boolean-expressions
+        if (
+            abs(fit_out[0] - 1.0) < 0.1
+            and abs(fit_out[2]) < 0.1
+            and reduced_chisq < 3
+            and (fit_err[0] is None or fit_err[0] < 0.1)
+            and (fit_err[1] is None or fit_err[1] < fit_out[1])
+            and (fit_err[2] is None or fit_err[2] < 0.1)
+        ):
+            return "computer_good"
+        else:
+            return "computer_bad"
+
+
+class T1Experiment(BaseExperiment):
+    """T1 experiment class"""
+
+    __analysis_class__ = T1Analysis
+
+    def __init__(
+        self,
+        qubit: int,
+        delays: Union[List[float], np.array],
+        unit: Optional[str] = "s",
+        experiment_type: Optional[str] = None,
+    ):
+        """
+        Initialize the T1 experiment class
+
+        Args:
+            qubit: the qubit whose T1 is to be estimated
+            delays: delay times of the experiments
+            unit:Optional, unit of the duration. Supported units: 's', 'ms', 'us', 'ns', 'ps', 'dt'.
+            experiment_type: Optional, the experiment type string.
+
+        Raises:
+            ValueError: if the number of delays is smaller than 3
+        """
+        if len(delays) < 3:
+            raise ValueError("T1 experiment: number of delays must be at least 3")
+
+        self._delays = delays
+        self._unit = unit
+        super().__init__([qubit], experiment_type)
+
+    # pylint: disable=arguments-differ
+    def circuits(self, backend: Optional["Backend"] = None) -> List[QuantumCircuit]:
+        """
+        Return a list of experiment circuits
+
+        Args:
+            backend: Optional, a backend object
+
+        Returns:
+            The experiment circuits
+
+        Raises:
+            AttributeError: if unit is dt but dt parameter is missing in the backend configuration
+        """
+
+        if self._unit == "dt":
+            try:
+                dt_factor = getattr(backend.configuration(), "dt")
+            except AttributeError as no_dt:
+                raise AttributeError("Dt parameter is missing in backend configuration") from no_dt
+
+        circuits = []
+
+        for delay in self._delays:
+            circ = QuantumCircuit(1, 1)
+            circ.x(0)
+            circ.barrier(0)
+            circ.delay(delay, 0, self._unit)
+            circ.barrier(0)
+            circ.measure(0, 0)
+
+            circ.metadata = {
+                "experiment_type": self._type,
+                "qubit": self.physical_qubits[0],
+                "xval": delay,
+                "unit": self._unit,
+            }
+
+            if self._unit == "dt":
```

Comment:
For the case of dt wouldn't it be better to convert "xval" to one of the standard time units rather than store it as dt? Then the analysis doesn't need to worry about handling dt as a special case.

Reply:
I agree that getting rid early from the peculiarity of dt is an advantage. On the other hand, this also creates some inconsistency that we may regret. For example, the user may expect to see in the metadata the unit and delays that she provided, and see in the fit's xrange the values that she provided.